### ISSUE TITLE
is that possible to give an output of predicting probability

### ISSUE THREAD

**Issue by zclecle2** [yuqcheng/scBalance#3]
Thanks for developing the easy-to-use package for annotating cells. I am wondering whether it is possible to give an output of predicting probability from the result.

**Comment by yuqcheng** [comment]
Hi, thanks for your attention on scBalance. It could give the predicting probability, which is the softmax layer output if the query data is loaded into it. I assume that you might want to obtain the confidence level or detect unknown cell types. However, I would not recommend this because our method is built only on a plain cross-entropy loss, which would lead to "over-confidence" (https://arxiv.org/abs/2106.04972) on the "unseen query sample" and thus the softmax output would not provide you with more information. 

We are developing a new package for scATAC-seq annotation named EpiPack (https://github.com/ZhangLabGT/EpiPack). In this package, we applied a new method for calculating the prediction prob. This could also be applied along with the result of scBalance. If you are interested in this, please refer to EpiPack. We are actively working on this package and will try to release it in 4 months. Thanks so much for your patience.

**Comment by zclecle2** [comment]
Thank you for the explanation for not recommending predicting the probability of unknown query data. Where I can find the softmax layer output if the query data is loaded into it? And I checked the EpiPack, may I ask that what function should I use for  calculating the prediction prob. Because I did not find the usage on your website.
And I found that if training data doesn't contain one cell type which my query data contains, it will assign a cell type in the training data instead of "unknown". So when we choose training data we should use one including all the cell types that our query data might contain. Thanks!

**Comment by yuqcheng** [comment]
Yes, you are correct. The model will assign a cell label even though it hasn't seen this cell type before. This is what I said "over-confidence" problem. This could be an OOD detection question but scBalance did not design this function. Thanks so much for checking the EpiPack repo. As I said the method is still under construction and you might see the full tutorials and packages in 3-4 months. At that time you may see how to use that OOD detection function. The paper is still under writing so please be patient and keep confidential of the code details in the EpiPack package. 

For the softmax output, I will update an output IO for you to obtain the prediction prob. Just be careful with the prob because of the overconfidence. I will do this in several days and let you know when I complete the updated version so that you can download it from PyPI. Thanks again for your attention on scBalance!

**Comment by zclecle2** [comment]
Many thanks to you for your reply and the work. The softmax output may be not that important to me as you said because of the overconfidence. So please take your time to update the package.  And now I am a bit confused about how to understand the predicting probability (softmax layer output). Or for the softmax layer output, when it is lower than what value, should I consider it feels not confident about its prediction. Again, much appreciated your efforts on this!
